dds/DCPS/ReactorTask.cpp

```diff
@@ -272,6 +323,42 @@ String ThreadStatus::get_key(const char* safety_profile_tid, const String& name)
   return key;
 }
 
+bool ThreadStatusManager::sync_with_parent(ThreadStatusManager& parent,
+  ThreadStatusManager::Map& running, ThreadStatusManager::Map& finished)
+{
+  ACE_WRITE_GUARD_RETURN(ACE_Thread_Mutex, g1, lock_, false);
+
+  {
+    ACE_READ_GUARD_RETURN(ACE_Thread_Mutex, g2, parent.lock_, false);
+
+    Map::iterator ci = map_.begin();
+    Map::iterator pi = parent.map_.begin();
+    bool got_child = ci != map_.end();
+    bool got_parent = pi != parent.map_.end();
```

Comment:
I'm not sure keeping these as separate variables is any better than just using comparisons with `end`, since the variables have to be updated at the right time.  I'm also not sure it's any worse.

Reply:
Yeah, they're left over from a version of the algorithm where I was doing more with the variables. Taking them away makes the `cmp` expression a bit harder to read, but I'm fine with that.